lib/response.js

```diff
@@ -1,58 +1,45 @@
 'use strict';
 
-/**
- * Constructor.
- */
-
-function Response(options) {
-  options = options || {};
+class Response {
+  constructor({ headers = {}, body = {}, ...otherOptions } = {}) {
+    this.status = 200;
+    this.body = body;
+    this.headers = {};
+
+    // Store the headers in lower case.
+    Object.entries(headers).forEach(([header, value]) => {
+      this.headers[header.toLowerCase()] = value;
+    });
+
+    // Store additional properties of the response object passed in
+    Object.entries(otherOptions)
+      .filter(([property]) => !this[property])
```

Comment:
I propose to use `Object.prototype.hasOwnProperty.call` here as well to avoid any tampering with the prototype chain